lib/hoststats/host.go

```diff
@@ -0,0 +1,97 @@
+package hoststats
+
+import (
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+var hostStatsCollectionInterval = 10 * time.Second
+
+// HostStats represents resource usage hoststats of the host running a Consul agent
+type HostStats struct {
+	Memory       *MemoryStats
+	CPU          []*CPUStats
+	DataDirStats *DiskStats
+	Uptime       uint64
+	Timestamp    int64
+}
+
+func (hs *HostStats) Clone() *HostStats {
+	clone := *hs
+
+	clone.CPU = make([]*CPUStats, len(hs.CPU))
+	for i := range hs.CPU {
+		cpu := *hs.CPU[i]
+		clone.CPU[i] = &cpu
+	}
+	return &clone
+}
```

Comment:
I think you can just do this:
```suggestion
func (hs *HostStats) Clone() *HostStats {
	clone := &HostStats{}
	*clone = *hs
	return clone
}
```

https://go.dev/play/p/1Uiyq3Dcq4m